server/index.js

```diff
@@ -50,6 +50,110 @@ app.get('/api/products/:id', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.get('/api/cart', (req, res, next) => {
+  if (!req.session) {
+    return (
+      res.status(400).json({
+        error: 'cart is empty'
+      })
+    );
+  }
+  const parameterizedArray = [req.session.cartId];
+  const sql = `
+  select  "c"."cartItemId",
+          "p"."image",
+          "p"."name",
+          "p"."price",
+          "p"."productId",
+          "p"."shortDescription"
+    from "cartItems" as "c"
+    join "products" as "p" using ("productId")
+  where "cartId" = $1;`;
+  db.query(sql, parameterizedArray)
+    .then(results => res.status(200).json(results.rows))
+    .catch(err => next(err));
+});
+
+app.post('/api/cart', (req, res, next) => {
+  const productId = parseInt(req.body.productId);
+  if (productId <= 0 || !productId) {
+    return res.status(400).json({
+      error: 'please input a valid product Id'
+    });
+  }
+
+  const parameterizedArray = [productId];
+  const sql1 = `
+  select "price"
+    from "products"
+    where "productId" = $1;`;
+  db.query(sql1, parameterizedArray)
+    .then(results => {
+      if (!results.rows.length) {
+        return next(new ClientError(`cannot find id at ${productId}`), 400);
```

Comment:
Calling `next()` does not prevent the rest of the promise chain from being executed. You need to `throw` the error instead of just passing it to `next()`.